datadog-opentelemetry/src/sampler.rs

```diff
@@ -205,6 +205,7 @@ mod tests {
         );
 
         // Clean up environment
+        #[allow(clippy::disallowed_methods)]
```

Comment:
You can just delete the line 209, the remove_var is a leftover from previous code we forgot to cleanup